diagnostics-app/src/app/app.component.it.spec.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Test file for the app.component.ts
+ */
+
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { AppComponent } from './app.component';
+import { Theme } from 'src/app/core/types/global.types';
+import { ThemeService } from './core/services/theme.service';
+
+describe('component: app', () => {
+  let fixture: ComponentFixture<AppComponent>;
+  let component: AppComponent;
+  let themeService: ThemeService;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [RouterTestingModule],
+      declarations: [AppComponent],
+      schemas: [NO_ERRORS_SCHEMA],
+      providers: [ThemeService],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AppComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+    themeService = TestBed.inject(ThemeService);
+  });
+
+  it('set cssClass on receiving update from themeService', () => {
+    expect(component.cssClass).toBe('dark');
+    let currentClass: Theme;
+
+    // subscribe to changes from themeService
+    // expect statements will run on receiving toggleUpdates
+    themeService.themeEventHandler.subscribe((next: Theme) => {
+      if (currentClass === 'dark') {
+        // gets executed when currentClass is dark
+        expect(next).toBe('light');
+      } else {
+        // gets executed when currentClass is light
+        expect(next).toBe('dark');
+      }
```

Comment:
It seems off to include logic in a test (even when it is simple). I think it is better to split the test into two (change dark mode to light mode, and vice versa). This will require setting the theme as a pre-test-work.

Or even better, to use parameterized test (if Angular supports it): https://rjzaworski.com/2013/01/parameterized-testing-in-javascript?

WDYT?

Reply:
Yeah it seemed off to me as well. This is the only way to test both cases inside a single function call. I went through the blog and yes, this is supported by Jasmine + Karma. I think when we have complicated logic later in the project, we can use parametrized testing.

For now, I think splitting this one into two tests makes sense because this is a simple one.